exoscale/instances_config.go

```diff
@@ -0,0 +1,83 @@
+package exoscale
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Instances configuration (<-> cloud-config file)
+type instancesConfig struct {
+	Disabled     bool // if true, disables this controller
+	Overrides    []instancesOverrideConfig
+	ExternalOnly bool `yaml:"externalOnly"` // if true, ignore Exoscale API (use only static overrides, if any)
+}
+
+type instancesOverrideConfig struct {
+	Name       string // considered a regexp if '/.../'
+	External   bool
+	ExternalID string `yaml:"externalID"`
+	Type       string
+	Addresses  []instancesOverrideAddressConfig
+	Region     string
+}
+
+type instancesOverrideAddressConfig struct {
+	Type    string // Hostname, ExternalIP, InternalIP, ExternalDNS, InternalDNS (v1.NodeAddressType)
+	Address string
+}
+
+// Return statically-configured instance override
+func (c *instancesConfig) getInstanceOverride(nodeName types.NodeName) *instancesOverrideConfig {
+	var config *instancesOverrideConfig
+
+	// first try an exact match on name
+	for _, candidate := range c.Overrides {
+		if len(candidate.Name) != 0 && nodeName == types.NodeName(candidate.Name) {
+			config = &candidate
+			break
+		}
+	}
+
+	// then regexp match on "name"
+	if config == nil {
+		for _, candidate := range c.Overrides {
+			if strings.HasPrefix(candidate.Name, "/") && strings.HasSuffix(candidate.Name, "/") {
+				if match, _ := regexp.Match(strings.Trim(candidate.Name, "/"), []byte(nodeName)); match {
```

Comment:
You should not ignore error here. When `err` is not nil, other return values are not expected to have a consistent value. Something like this would work:
```
r, err := regexp.Compile(strings.Trim(candidate.Name), "/")
if err != nil {
    continue // <= or return error if that makes more sense as regex expression is invalid
}
if r.MatchString(nodeName) {
    ....
```